lib/rules/no-get-by-assert.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const isDirectlyCalledByFunction = node =>
+  node.parent.parent.type === 'CallExpression';
+
+const isReturnedByArrowFunctionExpression = node =>
+  node.parent.parent.type === 'ArrowFunctionExpression';
+
+const isDeclared = node => node.parent.parent.type === 'VariableDeclarator';
+
+const isReturnedByReturnStatement = node =>
+  node.parent.parent.type === 'ReturnStatement';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow `getBy*` queries as assertion method itself',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('no-get-by-assert'),
+    },
+    messages: {
+      noGetByAssert: 'Disallowed use of `getBy*` query as implicit assert',
```

Comment:
Is this the message user will see on hover in VSCode for example? Should it include some information to use the `expect(queryBy*)...` pattern?

Reply:
Yes, this is the message you'll get when linting the code (or your IDE pass the linter for you). To be honest I don't know what to put here, so I just put something simple mirroring other rules from ESLint. Suggestions are more than welcome.

Reply:
I would go for something like `Use queryBy assertions instead of getBy`. What do you think?

Reply:
Same as previous mentions to `queryBy`: user can't be forced to do that, we can only mention the fact that the `getBy` must be transformed into explicit assert.